File: web/src/types/financial.ts
export interface FinancialProduct {
  id: string;
  product_type: 'input_financing' | 'crop_insurance' | 'market_linkage' | 'savings';
  name: string;
  description: string;
  provider: string;
  min_amount: number;
  max_amount: number;
  interest_rate_annual: number | null;
  commission_pct: number | null;
}

export interface FarmerRiskScore {
  farmer_id: string;
  score: number;
  components: Record<string, number> | null;
  last_computed_at: string | null;
}

export interface ProductEligibility {
  farmer_id: string;
  risk_score: number;
  eligible_products: FinancialProduct[];
  ineligible_reasons: Record<string, string> | null;
}

export interface AnalyticsSummary {
  total_queries: number;
  avg_cost_per_query_usd_cents: number;
  total_cost_usd_cents: number;
  avg_input_tokens: number;
  avg_output_tokens: number;
  avg_latency_ms: number;
  cache_hit_rate_pct: number;
}

File: app/models/financial.py
from datetime import datetime
from pydantic import BaseModel

from app.models.enums import FinancialProductType, TransactionStatus


class FinancialProduct(BaseModel):
    """A financial product available to farmers."""

    id: str
    product_type: FinancialProductType
    name: str
    description: str
    provider: str
    min_amount: float = 0.0
    max_amount: float = 0.0
    interest_rate_annual: float | None = None
    commission_pct: float | None = None
    eligible_farm_types: list[str] | None = None
    eligible_provinces: list[str] | None = None
    min_farm_size_ha: float = 0.0
    max_farm_size_ha: float = 100000.0
    active: bool = True


class FarmerRiskScore(BaseModel):
    """A farmer's computed risk score for financial product eligibility."""

    farmer_id: str
    score: int  # 0-100
    components: dict | None = None
    last_computed_at: datetime | None = None


class Transaction(BaseModel):
    """A financial product transaction record."""

    id: str
    farmer_id: str
    product_id: str
    transaction_type: TransactionStatus
    amount: float | None = None
    created_at: datetime | None = None


class ProductEligibility(BaseModel):
    """Result of checking a farmer's eligibility for financial products."""

    farmer_id: str
    risk_score: int
    eligible_products: list[FinancialProduct]
    ineligible_reasons: dict[str, str] | None = None
